src/cluster_legacy.c

```diff
@@ -7131,7 +7131,10 @@ int clusterCommandSpecial(client *c) {
         /* CLUSTER FORGET <NODE ID> */
         clusterNode *n = clusterLookupNode(c->argv[2]->ptr, sdslen(c->argv[2]->ptr));
         if (!n) {
-            if (clusterBlacklistExists((char *)c->argv[2]->ptr))
+            if (sdslen(c->argv[2]->ptr) != CLUSTER_NAMELEN) {
+                /* Sanity check for the provided node ID length*/
+                addReplyErrorFormat(c, "Bad node name %s", (char *)c->argv[2]->ptr);
```

Comment:
Maybe this is a poor error string?

Reply:
Why not just use the same error? I guess I would expect we would have just done `if (verifyClusterNodeId(c->argv[2]->ptr, sdslen(c->argv[2]->ptr)) && clusterBlacklistExists((char *)c->argv[2]->ptr)) {`

Reply:
Actually, maybe we do a length check in clusterBlacklistExists? Then we also cover the other location? Seems a bit more defensive.

Reply:
Well. I thought of that but was trying to avoid changing the API and provide a way to distinguish between the case the node was not found and the fact that the user made a mistake (like forgot to add the last character). However I am fine changing the   clusterBlacklistExists API to accept length as well. just +1 on this if you prefer it that way

Reply:
adding that it makes it easier to add test :)

Reply:
return a `Unknown node` seem better since we has other places like this, like we also don't do the length check in this case
```
    } else if (!strcasecmp(c->argv[3]->ptr, "node") && c->argc >= 5) {
        /* CLUSTER SETSLOT <SLOT> NODE <NODE ID> */
        n = clusterLookupNode(c->argv[4]->ptr, sdslen(c->argv[4]->ptr));
        if (!n) {
            addReplyErrorFormat(c, "Unknown node %s", (char *)c->argv[4]->ptr);
            return 0;
        }
```

Reply:
Yeah. I agree with Binbin. We've discussed in the past having alternative node names as well, I don't want to couple the length to the format of the node name.

Reply:
@madolson @enjoy-binbin fixed per your comments.